### ISSUE TITLE
Unable to install Ubuntu Touch on Nexus 7 flo

### ISSUE THREAD

**Issue by Stefano0101** [ubports/ubuntu-touch#300]
- Device: Nexus 7 flo
- Channel: 15.04/Stable
- Build: Ota-1

## Steps to reproduce

1. I have tried to flash Ubuntu Touch on Nexus 7 flo, but failed. Ubuntu-device-flash failed, MDT failed, Ubports-installer failed
2. Error I get are shown down on the pictures and Terminal output
3. Possibly related to this: https://askubuntu.com/questions/674179/ubuntu-device-flash-fails-on-nexus-7-2013-android-5-0-2-cant-copy-image-to/675499

## Expected behavior

Just to flash it and enjoy it, but for now just a lot of hours wasted and no solution found :-)

## Actual behavior

I HAVE 3 NEXUS 7 TABLETS AND THIS IS HAPPENING ON ALL OF THEM.

## Logfiles and additional information

I have downloaded the files forehand and also tried direct flash to the device with the [same result](https://user-images.githubusercontent.com/29058699/31634760-820246b4-b2bc-11e7-91cd-126d5b3a5505.jpg).

```
ubuntu-device-flash --server=https://system-image.ubports.com touch --device=flo --channel=ubports-touch/15.04/stable --bootstrap
2017/10/16 20:47:45 Device is |flo|
2017/10/16 20:47:45 Flashing version 1 from ubports-touch/15.04/stable channel and server https://system-image.ubports.com to device flo
2017/10/16 20:47:52 Waiting for device to enter recovery mode ...
2017/10/16 20:48:02 Start pushing /home/stefano/.cache/ubuntuimages/pool/ubports-03a48725ab23059f103ccbb3cfa913f1ebcd314aaa0742e9ff724ee4fef19d78.tar.xz to device
2017/10/16 20:48:02 Start pushing /home/stefano/.cache/ubuntuimages/pool/keyring-4c4e7ef380ebcfa2c31084efa199138e93bfed8fc58aa3eb06bdf75a78af9b57.tar.xz to device
2017/10/16 20:48:02 Start pushing /home/stefano/.cache/ubuntuimages/ubports-touch/15.04/stable/flo/version-1.tar.xz to device
2017/10/16 20:48:02 Start pushing /home/stefano/.cache/ubuntuimages/pool/device-d7801a4ea82350f68befeb2a02315a56068784b5d3f0c1d5396f4e02b6dd3719.tar.xz to device
2017/10/16 20:48:02 Start pushing /home/stefano/.cache/ubuntuimages/gpg/image-master.tar.xz to device
2017/10/16 20:48:02 Start pushing /home/stefano/.cache/ubuntuimages/gpg/image-signing.tar.xz to device
2017/10/16 20:48:02 error pushing: failed to copy '/home/stefano/.cache/ubuntuimages/gpg/image-signing.tar.xz' to '/cache/recovery/': Is a directory
```

**Comment by Stefano0101** [comment]
Btw, I've tried to flash Canonical image as well, same result, so probably is the same as here:
https://askubuntu.com/questions/674179/ubuntu-device-flash-fails-on-nexus-7-2013-android-5-0-2-cant-copy-image-to/675499

**Comment by APokorny** [comment]
If you look at 
https://github.com/ubports/android_kernel_google_msm/blob/ubp-5.1/drivers/mmc/core/mmc.c#L296
the change is already in place. So the error is somewhere else

**Comment by Stefano0101** [comment]
I downgraded the tablet to 4.4 kitkat base and tried to flash it, just in case if the firmware downgrade changed the paths/partitions ir whatever that is and the tablet won't even boot up! I mean Android. Factory reset in recovery gives similar error logs, interesting.

See the photo.
![image20171017_125119560](https://user-images.githubusercontent.com/29058699/31666350-cc8f07bc-b343-11e7-85d4-a9e62d829ab3.jpg)

**Comment by Stefano0101** [comment]
There is only version-1 on server, stable channel.
https://system-image.ubports.com/ubports-touch/15.04/stable/flo/

**Comment by Arc676** [comment]
That is weird. Having checked again, the 'About' section in System Settings says "OTA-15". I'm not exactly sure what that means, but I do remember UBPorts announcing support for the Nexus 7 with the release of OTA-2.

Just to be sure, you used `git clone` to get MDT, right?

**Comment by Stefano0101** [comment]
Yes, git cloned to get MDT. And MDT working well. Btw, if your System Setting says Ota-15 , you are still on Canonical image, I think. If you were on Ubports it should say r1 or whatever  Ota it is for given device. Pro 5 is on OTA-2 and the OS build number is r2.
Anyway, I even tried the Canonical image, OTA-13 as is the last stable relase and same result.

**Comment by Arc676** [comment]
That's interesting. It would appear that the two methods for obtaining Ubuntu Touch for flo as presented [here](https://devices.ubports.com/#/flo) are actually different.

Using the `system-image` server, the command is
```
sudo ubuntu-device-flash --server=http://system-image.ubports.com touch --device=flo \
--channel= --bootstrap --recovery-image=
```

However, using MDT, the command that is run by the script is 
```
ubuntu-device-flash touch --channel ubuntu-touch/stable/ubuntu --device flo --bootstrap
```

**Comment by Stefano0101** [comment]
Label question?

**Comment by Stefano0101** [comment]
Anyone have similar problem? Anyone to help?

**Comment by rubencarneiro** [comment]
[log.txt](https://github.com/ubports/ubuntu-touch/files/1408588/log.txt)

First you should remove the ubuntuimages dir rm -rf .cache/ubuntuimages
Then you should try again and it will work.
![log](https://user-images.githubusercontent.com/6282290/31910075-3fd5741e-b834-11e7-87af-a40025ab002a.png)

**Comment by Stefano0101** [comment]
Did you mean to remove .cache/ubuntuimages from my PC?
Ihave done it many times to clear the old files, yet the same happening.
If you look at the screenshot, the log is saying that  mmsev/block/msm_sdcc.1/by-name/misc  is missing, no such a directory.

And the other one says  can't copy to .cache/recovery   is a directory  I would have to rm -rf  recovery directory on phone.

**Comment by Stefano0101** [comment]
I have done it and interesting thing happen. After I created /cache/recovery  directory, the  files got pushed to the device, rebooted to recovery and exactly same logs appeared.
![1508067173567-1508067145550](https://user-images.githubusercontent.com/29058699/32008522-4b2738e0-b9a4-11e7-96ab-73247cfebe91.jpg)


 Nothing happens after.I rebooted the tablet and it booted to Android like it wasn't wiped at all.


ubuntu-device-flash --server=https://system-image.ubports.com touch --device=flo --channel=ubports-touch/15.04/stable --wipe
2017/10/25 16:28:29 Device is |flo|
2017/10/25 16:28:29 Flashing version 1 from ubports-touch/15.04/stable channel and server https://system-image.ubports.com to device flo
2017/10/25 16:28:29 Start pushing /home/stefano/.cache/ubuntuimages/gpg/image-master.tar.xz to device
2017/10/25 16:28:29 Start pushing /home/stefano/.cache/ubuntuimages/gpg/image-signing.tar.xz to device
2017/10/25 16:28:29 Start pushing /home/stefano/.cache/ubuntuimages/pool/keyring-4c4e7ef380ebcfa2c31084efa199138e93bfed8fc58aa3eb06bdf75a78af9b57.tar.xz to device
2017/10/25 16:28:29 Start pushing /home/stefano/.cache/ubuntuimages/ubports-touch/15.04/stable/flo/version-1.tar.xz to device
2017/10/25 16:28:30 Done pushing /home/stefano/.cache/ubuntuimages/gpg/image-master.tar.xz to device
2017/10/25 16:28:30 Done pushing /home/stefano/.cache/ubuntuimages/pool/keyring-4c4e7ef380ebcfa2c31084efa199138e93bfed8fc58aa3eb06bdf75a78af9b57.tar.xz to device
2017/10/25 16:28:30 Done pushing /home/stefano/.cache/ubuntuimages/gpg/image-signing.tar.xz to device
2017/10/25 16:28:30 Done pushing /home/stefano/.cache/ubuntuimages/ubports-touch/15.04/stable/flo/version-1.tar.xz to device
2017/10/25 16:28:30 Start pushing /home/stefano/.cache/ubuntuimages/pool/device-d7801a4ea82350f68befeb2a02315a56068784b5d3f0c1d5396f4e02b6dd3719.tar.xz to device
2017/10/25 16:28:31 Start pushing /home/stefano/.cache/ubuntuimages/pool/ubports-03a48725ab23059f103ccbb3cfa913f1ebcd314aaa0742e9ff724ee4fef19d78.tar.xz to device
2017/10/25 16:28:43 Done pushing /home/stefano/.cache/ubuntuimages/pool/device-d7801a4ea82350f68befeb2a02315a56068784b5d3f0c1d5396f4e02b6dd3719.tar.xz to device
2017/10/25 16:29:17 Done pushing /home/stefano/.cache/ubuntuimages/pool/ubports-03a48725ab23059f103ccbb3cfa913f1ebcd314aaa0742e9ff724ee4fef19d78.tar.xz to device
2017/10/25 16:29:17 Created ubuntu_command: /home/stefano/.cache/ubuntuimages/ubuntu_commands044462214
2017/10/25 16:29:17 Rebooting into recovery to flash

**Comment by Stefano0101** [comment]
Nobody else has such a problem to install it? Wha's wrong with it, I don't get it?

**Comment by WammKD** [comment]
I've had this problem, back when I first installed UT on my Nexus 7 (actually, went through the process twice). I was hoping I still had those files lying around (I had a different computer, at the time) but I've searched both my current computer and every file on my external hard drive and no luck, I'm afraid.

I came across this thread (for Deb, sadly): https://drive.google.com/file/d/0ByR5uBE8p7qmYnlqb1pNZl90ZXc/view?usp=sharing. But the original AskUbuntu answer gives us the exact diff that makes the difference and it's even only one line so I figured I could try following the directions at https://developer.ubuntu.com/en/phone/devices/porting-new-device/ and just edit the appropriate kernel file myself and build new boot.img and recovery.img files (assuming I understand how things work, correctly).

Unfortunately (still), I couldn't even build anything when following the directions vanilla. I ran `phablet-dev-bootstrap phablet`, `. build/envsetup.sh`, and then `lunch` but it never produced an **out** directory. I ran `make -j4`, anyway, and it printed things out but didn't seem to add or change any files.

If someone else wants to take a swing at the same process, perhaps they might have better luck than I.

**Comment by Stefano0101** [comment]
So this surely looks like a bug in the Flo image or the installer(u-d-f or MDT or Ubports Installer).
Anyone can make a TWRP backup of a clean Ubuntu Touch/Ubports install? I can try to just restore it and it might work. PLEASE?????

**Comment by Arc676** [comment]
Um... I have literally no idea what's wrong with your device. Immediately before I began to write this comment, I successfully installed UBPorts' Ubuntu Touch on my Nexus 7 (flo).

OS details now:
OS: Ubuntu 15.04 (r1)
Build number: r1 (no idea what that means)

Machine I used to install:
MacBook Pro, running Ubuntu 16.04 LTS

I used the `AppImage` file from [here](https://github.com/ubports/ubports-installer/releases/tag/0.1.7-beta)

**Comment by Stefano0101** [comment]
But with 3 tablets ??? Same happening on all of them.

**Comment by Stefano0101** [comment]
I have tried to flash it on a different Ubuntu PC, same result, driving me nuts.

**Comment by rubencarneiro** [comment]
Nexus 7 as seriuos problem with some usb cables, try another cable and retry

**Comment by WammKD** [comment]
At what point did you buy your Nexus 7, @Arc676? According to the answer in the AskUbuntu link, "newer Nexus 7 'flo' models, made in/after late 2014 and shipped with Android L have a different revision to their eMMC controller/hardware/something".

I mostly give so much credit to that answer since I first bought my Nexus 7 after 2014 when I first wanted to try installing Ubuntu Touch and ran into this issue. Both times I tried installing UT on my device, I found myself having to use the boot.img and recovery.img provided by `ddagunts` and that solved my issue in terms of being able to install Canonical's Ubuntu Touch.

He provides the patch change in question (I believe) in the AskUbuntu answer; if we could build a new boot.img and recovery.img with a new kernel with that patch, I think it would solve @Stefano0101's and my issue.

**Comment by Arc676** [comment]
@WammKD I bought this second-hand Nexus off the internet this year, but who knows when the original owner bought it.

In case they're relevant to determining the version/date, fastboot gives these details:
Product: flo
Variant: flo 32G
HW: rev_e
Bootloader version: FLO-0.4.05
Carrier info: none (obviously b/c it's the wifi model)
Serial number: (not relevant)
Signing: yes
Secure boot: enabled
Lock state: unlocked

**Comment by WammKD** [comment]
Makes sense.

I don't know if it'll be any useful; my fastboot give me:
PRODUCT NAME – flo
VARIANT – flo 16G
HW VERSION – rev_e
BOOTLOADER VERSION – FLO-04.05 (I don't know if it's relevant but it displays 04.05 on mine as opposed to 0.4.05)
CARRIER INFO – None
SERIAL NUMBER – (not relevant)
SIGNING – yes
SECURE BOOT – enabled
LOCK STATE – unlocked

Pretty similar to yours.
I really wish I still had those boot.img and recovery.img files, still, and then I could just try flashing them and see directly if that's the solve, like last time.

**Comment by WammKD** [comment]
You know, it just dawned on me – @Arc676 –, – if the issue really is a kernel difference – it might make more sense to compare kernel versions?

What's the result of `adb shell cat /proc/version` for you? Mine spits out:

`Linux version 3.4.0-5-flo (root@meow) (gcc version 4.7.3 (Ubuntu/Linaro 4.7.3-11ubuntu1) ) #19 SMP PREEMPT Thu Sep 17 17:38:43 PDT 2015`

**Comment by Arc676** [comment]
That was a typo, mine also says 04.05 without through first dot. My bad. 

`cat /proc/version` gave me
```
Linux version 3.4.0-5-flo (buildd@kishi14) (gcc version 4.7.4 (Ubuntu/Linaro 4.7.4-3ubuntu2) ) #1-Ubuntu SMP PREEMPT Thu Sep 3 14:21:37 UTC 2015
```

**Comment by WammKD** [comment]
Haha; no worries.

Welp, that tells us nothing, I don't think….

**Comment by Stefano0101** [comment]
I have tied a different usb cable as well, from N5 and I know this one is good working one and the same as before happens.
COULD I ASK YOU FOR A FAVOUR??? If you flash your N7 with TWRP recovery and create a TWRP backup that could be shared with me, it might just work. Then you can always reflash Ubuntu recovery and continue to use the tablet as nothing happens, but I might be able to by-pass this installation error. I have done similar thing for other users and shared the Pro 5 backup with them when they had a errors and problem to install it and this way it worked well.
PLEASE, ANYONE, SHARE A BACKUP of Ubports or Canonical image with me!!!!!!!!!!!!!

**Comment by Arc676** [comment]
Is TWRP recovery part of the recovery thing? Or do you just need `/userdata/ubuntu.img` or `/userdata/system.img`?

**Comment by Stefano0101** [comment]
no TWRP is not part of recovery thing, but it's easier just to restore the backup of whole device. TWRP then formats and restores all partitions and it should just work.

http://www.devfiles.co/download/XJ2i3IOU/twrp-3.0.0-0-flo.img

**Comment by Stefano0101** [comment]
I would need both, system.img and ubuntu.img if possible, please?

**Comment by Arc676** [comment]
Does TWRP have a function for only backing up certain parts of the device? Obviously a backup that becomes available online should not include sensitive information such as passwords.

**Comment by Stefano0101** [comment]
Off course, yes you can back up only certain partitions, that's possible to choose from the list.

The backup needs to be done from a fresh install or factory reseted device from within Ubuntu Touch OS,then flashed with TWRP recovery and creating the backup itself, then compressed and it's ready to share. There should be no personal information, files, passwords or anything else that could be compromising anyone. I

****It's good to first flash  TWRP**, make a backup of the Ubuntu Touch installation for later restoring the device to previous state with all the data's, passwords, apps, etc... and flashing Ubuntu Touch recovery.img after restoring the device in order to receive the OTA. (and also is handy to have Ubuntu Touch backup stored on PC, in case something goes wrong in the future )**

The backup we talking here about is just plain, nothing is there, it boots to setup.

**Comment by Stefano0101** [comment]
still haven't figured it out, nothing works...

**Comment by Stefano0101** [comment]
Nobody to help with this?

**Comment by Stefano0101** [comment]
It looks like there is not many Nexus 7 Ubuntu Touch users. Pity, it would be nice to have a 7'' tablet with UT. 
Anyway, I give up on this one, unless Ubports fix the issue.

**Comment by Flohack74** [comment]
Hi, the only thing I can do here is to remove the question flag, as it is clearly not aquestion anymore :) and lets hope someone picks it up. I personally do not have one of those devices.

Just in case, could you think shipping us one of them for testing?

BR

**Comment by Stefano0101** [comment]
I could, if it comes back later :-) (in best case scenario with UT installed on it :-) )
Need the address for sending though.

**Comment by Flohack74** [comment]
Yeah I will try to find a developer interested in hacking into this. Maybe @APokorny ? =)

**Comment by rubencarneiro** [comment]
see here works ok, no error https://www.youtube.com/watch?v=ll8Gut8KJl0&feature=youtu.be

**Comment by Stefano0101** [comment]
rubencarneiro, 
Any chance that you flash TWRP recovery and make a backup that you can share with me? Just a fresh install, no  personnal information, no pasword, no data from your device? I think it might work, but nobody offered to do it for me and possibly for other UT Nexus 7 users. Could you help?

**Comment by rubencarneiro** [comment]
Ok i will make a backup

**Comment by rubencarneiro** [comment]
just extract and copy to BACKUP folder on team win https://drive.google.com/file/d/198K3g2MaHSWMxECRO9yKoawkhnn_5zAP/view?usp=sharing

**Comment by Stefano0101** [comment]
rubencarneiro
Thanks for the backup, but sadly it did not work. It restores the backup without any issue, but tablet won't boot at all. Thanks for trying to help.

**Comment by rubencarneiro** [comment]
Stefano0101 maybe when you want we could connect trough teamviewer and let me try a thing or two

**Comment by hoh61** [comment]
Hello Stephano,

don't give up too fast. When I heard that Ubuntu will suspend their activities on ubuntu touch I was worrying about the future of my M10 FHD tablet. It's a wonderful hardware thing; but without software support? I faced this already in the past for the Nexus7, where I installed an UBUNTU image, which was not further supported shortly after that. However, I developed some applications on that device which are still running.

So my expectation is, that the 16.04 version will be soon available for the flo too. In the meantime I will upgrade my application to work with 16.04 and (hopefully) support the development of the 16.04 target with a lot of issues/bugs ...

**Comment by NeoTheThird** [comment]
Duplicate of #643.

**Comment by UniversalSuperBox** [comment]
This is not About the 'deb' device, but rather about 'flo'. We had another person hit this error today and we were able to track down the issue to what @Stefano0101 found: https://askubuntu.com/questions/674179/ubuntu-device-flash-fails-on-nexus-7-2013-android-5-0-2-cant-copy-image-to/675499

We will need to rebuild the boot image and recovery for the Nexus 7 to fix this issue. That isn't a simple fix since we need to rebuild the kernel out-of-tree (we're using Canonical's build of Android and Linux) as @APokorny discovered with the Nexus 4.

**Comment by UniversalSuperBox** [comment]
Let's try that again. Same instructions as above, new files.

[flo-newkernel-2.zip](https://github.com/ubports/ubuntu-touch/files/2715416/flo-newkernel-2.zip)
[flo-newkernel-2.zip.asc.txt](https://github.com/ubports/ubuntu-touch/files/2715417/flo-newkernel-2.zip.asc.txt)

```
$ sha512sum flo-newkernel-2.zip
4692a2bf6045692bf0eda1f43a80db3463a0a1da642cf10b33aa67e1c5d6de699ef7d40766a78cade8c3854134b5a85ac2f010951853a3cb15147c5690095349  flo-newkernel-2.zip
```

**Comment by UniversalSuperBox** [comment]
Today's devel image has a new kernel module built with the patch that @Stefano0101 found. Anyone who was having trouble installing Ubuntu Touch on their Nexus 7, you can use the installer:

* Select Devel today through Wednesday the 2nd when the next RC image is released (or earlier if we do a call for testing)
* Select RC after its next release
* Select Stable after OTA-7

**Comment by UniversalSuperBox** [comment]
@Stefano0101, have you been able to install Ubuntu Touch on one of your three Nexus 7 2013's?

**Comment by Stefano0101** [comment]
Hi Dalton, I gave away 2 of them and the last one is using my wife with Android on it. ( I might try to flash it though) Thinking about getting another N7, if I come across this particular problem, I'll report back. Thanks for fixing this bug anyway.

**Comment by UniversalSuperBox** [comment]
That's unfortunate, but understandable given the age of this bug. I'll move it to "Done" since I've had one confirmation that the fix works. We can revisit if we get a report otherwise.

**Comment by wetheral** [comment]
PRODUCT NAME – flo
VARIANT – flo 16G
HW VERSION – rev_e
BOOTLOADER VERSION – FLO-04.05
CARRIER INFO – None
SERIAL NUMBER – (not relevant)
SIGNING – yes
SECURE BOOT – enabled
LOCK STATE – unlocked

Many thanks for all the work (@UniversalSuperBox) & guidance here. I have just successfully installed a flo. 

**Working from ubuntu 16.04 appears to be a must.** - did not work from W10 or U18.04 
#1
Followed these instructions from above:
> Anyone who has a Nexus 7 affected by this change, I've rebuilt the boot and recovery images. I've attached them along with a GPG signature from myself. Please test these to see if they work.
> 
> To use these, you'll need to use `ubuntu-device-flash` rather than the installer. Use this command:
> 
> ```
> ubuntu-device-flash touch --server=https://system-image.ubports.com --device=flo ----recovery-image=recovery.img
> ```
> After the install is completed, you may need to install the new boot and recovery images with `fastboot flash boot boot.img` and `fastboot flash recovery recovery.img`. You may also need to do this after system-image updates until this is placed on our system-image server and this issue is closed.
> 
> You can install ubuntu-device-flash in Ubuntu 16.04 from scratch, or you can download [this deb](https://repo.ubports.com/pool/bionic/main/g/goget-ubuntu-touch/ubuntu-device-flash_0.35+ubports+0~20181221200828.5~1.gbpd7daaa_amd64.deb) and install it. It doesn't have any other requirements from our servers.
> 

**..... note updated post/files**
> Let's try that again. Same instructions as above, new files.
> 
> [flo-newkernel-2.zip](https://github.com/ubports/ubuntu-touch/files/2715416/flo-newkernel-2.zip)
> [flo-newkernel-2.zip.asc.txt](https://github.com/ubports/ubuntu-touch/files/2715417/flo-newkernel-2.zip.asc.txt)
> 
> ```
> $ sha512sum flo-newkernel-2.zip
> 4692a2bf6045692bf0eda1f43a80db3463a0a1da642cf10b33aa67e1c5d6de699ef7d40766a78cade8c3854134b5a85ac2f010951853a3cb15147c56
> ```
#2. 
Had issue whilst flashing(with cache partion I beleive) system hung on Google screen. Rebooted to (ubuntu) recovery - factory reset, cleared cache. Then repeated #1. - this time with no error.

Hope this helps

**Comment by gmelchett** [comment]
Executing:
`ubuntu-device-flash touch --server=https://system-image.ubports.com --device=flo ----recovery-image=recovery.img`
results in:
`2019/02/23 11:02:25 Device is |flo|
Channel ubuntu-touch/stable/ubuntu not found on server https://system-image.ubports.com`

Any idea how to change to the RC channel?

**Comment by gmelchett** [comment]
No success. I'm still stuck at the chip-stop-sign :-(
I tried ` ubuntu-device-flash touch --device=flo --server=https://system-image.ubports.com --channel=ubports-touch/16.04/rc --recovery-image=flo-recovery-new.img`
I use with the recovery image from the zip-file mentioned a few posts up.

**Comment by UniversalSuperBox** [comment]
There should be no need to use these instructions any more. Please try to use the UBports Installer (https://devices.ubuntu-touch.io/device/flo) to install Ubuntu Touch on your device.

**Comment by gmelchett** [comment]
Installing via the UBports installer version 0.1.21 beta does unfortunately not work - for me. My flo-tablet is stuck at the "Stop sign".
I tried to install via the UBports installer first, before reading this bugreport, but I did try again after your comment. I reinstalled the UBports installer and did a fresh "flash-all" with Googles official images and script.
When trying to install UBports, I selected "Wipe" and "16.04/rc".

**Comment by bmw2621** [comment]
@gmelchett is correct, I have been having the same problem for months, trying everything in this thread.

The installer does not work on flo (or at least a version I have.)

I was able to get the ubuntu-device-flash option to go further by adding the --bootstrap tag and installing over the failed UBports-installer install.  Now I cant get past

**Comment by Flohack74** [comment]
Please open new issues, and don't comment on closed ones. Nothing will happen here any more, as we do not see closed bugs as needing attention.

**Comment by cmmei** [comment]
Ubuntu 18.04 does not install `ubuntu-device-flash` .

**Comment by r2d2agr** [comment]
You have to reflash the factory image many times

**Comment by cmmei** [comment]
```
root@admin:~# sudo ubuntu-device-flash --server=https://system-image.ubports.com touch --device=flo --channel=ubports-touch/16.04/stable --bootstrap --wipe
2019/07/21 02:12:17 Device is |flo|
2019/07/21 02:12:20 Flashing version 7 from ubports-touch/16.04/stable channel and server https://system-image.ubports.com to device flo
61.89 MB / 61.89 MB [======================================] 100.00 % 1.57 MB/s 
371.72 MB / 371.72 MB [====================================] 100.00 % 2.00 MB/s 
```
It does not start to do anything when downloaded flashing version by `ubuntu-device-flash`.

**Comment by cmmei** [comment]
It `ubuntu-device-flash` works well. I switch my device to recovery mode by pushing power with vol down. Then It starts pushing.

[Ubuntu_16.04-install UT in Nexus7_2013_wifi_terminal.txt](https://github.com/ubports/ubuntu-touch/files/3413945/Ubuntu_16.04-install.UT.in.Nexus7_2013_wifi_terminal.txt)

**Comment by UniversalSuperBox** [comment]
My previous information was incorrect, and I'm sorry it took this long to get it fixed.

We have now adjusted the installer so it downloads the new boot and recovery images compiled with the patch mentioned in this post. Previously these images were getting used in the final Ubuntu Touch install, but weren't used for the install process. Now this has been fixed.

You should be able to install on your newer Nexus 7 2013's with the UBports Installer.

**Comment by ploum** [comment]
I tried to install with the MacOS installer today on a Nexus 7 wifi 2013. The installer told me everything was fine but, at reboot, the device is now stuck on "This phone needs restoring from a PC or a service center"

Tried 16.04/release and devel without any change.

![2019-08-27 13 17 21](https://user-images.githubusercontent.com/1233155/63768054-87953980-c8cf-11e9-8e74-eace24e06fc0.jpg)

**Comment by valdesvj** [comment]
Sorry to write here but It looks that the issue is closed but  yesterday during this crazy isolation due covid-19, may be someone has the same idea and try to get back in business a Nexus 7 installing ubuntu. 

So trying to install ubuntu touch on my Nexus 7 (flo) and got same problem, stuck in the recovery mode of ubuntu.

Going  through adb shell from Archlinux.

Checked size of cache it was only 74MB, dismounted cache and data, deleted the partitions with parted command and created again, this time we gave the cache data 1GB.

After that gave a new try with the ubuports-app and managed to get Ubuntu touch in my nexus 7.

On the nexus 7, cache was 23 and userdata was on 31, when you deleted the partitions and create again they are getting the same number, but between commands check in parted with the print command that everything is going as it shloud be 

-------
adb shell
umount cache
umount data
parted
rm 23
rm 31
mkpartfs extended ext4 2611MB 3611MB
name 23 cache
mkpartfs extended ext4 3611MB 27GB
name 31 userdata
exit
exit
-------

Happy Nexus 7
